app/src/main/java/com/example/babytracking/BreastActivity.kt

```diff
@@ -1,11 +1,136 @@
 package com.example.babytracking
 
 import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.TimePicker
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.breast_layout.breast_quality_baby
+import kotlinx.android.synthetic.main.breast_layout.breast_button_save
+import kotlinx.android.synthetic.main.breast_layout.breast_length
+import kotlinx.android.synthetic.main.breast_layout.breast_quality_mother
+import kotlinx.android.synthetic.main.breast_layout.spinner_breast
+
+class BreastActivity : AppCompatActivity() {
+
+    private var hour: Int = 0
+    private var minute: Int = 0
+    private var ampm: String = "AM"
+    private var breast: String = ""
 
-class BreastActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.breast_layout)
+
+        breastTimeClick()
+        breastSelection()
+        breastSave()
+    }
+
+    private fun breastTimeClick() {
+        val timePicker = findViewById<TimePicker>(R.id.sleep_timepick_hour)
+        timePicker.setOnTimeChangedListener { _, pickerHour, pickerMinute ->
+            hour = pickerHour
+            minute = pickerMinute
+
+            // AM_PM decider logic
+            when {
+                hour == 0 -> {
+                    hour += 12
+                    ampm = "AM"
+                }
+                hour == 12 -> ampm = "PM"
+                hour > 12 -> {
+                    hour -= 12
+                    ampm = "PM"
+                }
+                else -> ampm = "AM"
+            }
+        }
+    }
+
+    private fun breastSelection() {
+        if (spinner_breast != null) {
```

Comment:
Can this even be null? Why?

If it's truly an issue, consider returning early to avoid having too much nested code:

```kt
if (spinner_breast == null) {
  return
}
```

Or you could try using the Elvis `?:` operator:

```kt
spinner_breast ?: return
```